### PR TITLE
Improve handling of introspection properties.

### PR DESCRIPTION
**Pull request by paulbakker**
Fix for #2024.

* Setting the property no longer causes a duplicate bean 
* Catch the corner case of setting both properties 
* Either property can be used

Pull Request type
----

- [ x] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe):

### CODE REVIEW

File: graphql-dgs-spring-boot-oss-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/autoconfig/DgsAutoConfiguration.kt [hunk 1]
```diff
@@ -289,7 +289,9 @@ open class DgsAutoConfiguration(
             Duration.parse(configProps.preparsedDocumentProvider.cacheValidityDuration),
         )
 
+    // TODO: Remove when legacy modules are removed. This is also handled in DgsSpringGraphQLAutoConfiguration
```
Comment (iparadiso) [inline graphql-dgs-spring-boot-oss-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/autoconfig/DgsAutoConfiguration.kt:292]: nit:  consider adding github issues with any TODO comments to more easily trace your way back to this code

**Review by iparadiso — APPROVED** [review]

**Comment by paulbakker** [comment]
Good idea, will do

**Review by srinivasankavitha — APPROVED** [review]

File: graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/autoconfig/DgsSpringGraphQLEnvironmentPostProcessor.kt [hunk 2]
```diff
@@ -28,7 +29,19 @@ class DgsSpringGraphQLEnvironmentPostProcessor : EnvironmentPostProcessor {
     ) {
         val properties = mutableMapOf<String, Any>()
 
-        properties["spring.graphql.schema.introspection.enabled"] = environment.getProperty("dgs.graphql.introspection.enabled") ?: true
+        if (environment.getProperty("spring.graphql.schema.introspection.enabled") != null &&
```
Comment (iuliiasobolevska) [inline graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/autoconfig/DgsSpringGraphQLEnvironmentPostProcessor.kt:32]: consider using local vars or better constants for property names (idk what's the probability of ever having to rename `spring.graphql.schema.introspection.enabled` but updating 4 strings just in this file is error-prone)

**Review by iuliiasobolevska — APPROVED** [review]

File: graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/autoconfig/DgsSpringGraphQLEnvironmentPostProcessor.kt (hunk 2 above)
Comment (paulbakker) [reply graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/autoconfig/DgsSpringGraphQLEnvironmentPostProcessor.kt]: done

**Pull request merged**